code/controllers/subsystem/vis_overlays.dm

```diff
@@ -89,10 +93,81 @@ SUBSYSTEM_DEF(vis_overlays)
 	var/list/overlays_to_remove = list()
 	for(var/i in thing.managed_vis_overlays - unique_vis_overlays)
+		if(istype(i, /obj/effect/overlay/vis/mob_alpha))
+			continue
 		var/obj/effect/overlay/vis/overlay = i
 		add_vis_overlay(thing, overlay.icon, overlay.icon_state, overlay.layer, overlay.plane, turn(overlay.dir, rotation), overlay.alpha, overlay.appearance_flags)
 		overlays_to_remove += overlay
 	for(var/i in thing.managed_vis_overlays & unique_vis_overlays)
+		if(istype(i, /obj/effect/overlay/vis/mob_alpha))
+			continue
 		var/obj/effect/overlay/vis/overlay = i
 		overlay.dir = turn(overlay.dir, rotation)
 	remove_vis_overlay(thing, overlays_to_remove)
+
+
+/// just add_vis_overlay() but only visible to observers
+/datum/controller/subsystem/vis_overlays/proc/add_obj_alpha(atom/movable/thing, icon=null, iconstate=null, alpha=150)
+	add_vis_overlay(thing, icon || thing.icon, iconstate || thing.icon_state, thing.layer+0.1, thing.plane, thing.dir, alpha=alpha, add_appearance_flags=RESET_ALPHA, invisibility=INVISIBILITY_OBSERVER)
+
+/// identical to procs above, but mob alpha version
+/datum/controller/subsystem/vis_overlays/proc/add_mob_alpha(atom/movable/thing, mob/target_mob, alpha=150, invisibility=INVISIBILITY_OBSERVER)
+	var/obj/effect/overlay/vis/mob_alpha/overlay
+	var/mob_alpha_id = "[REF(target_mob)][alpha][invisibility]"
+	overlay = mob_alpha_vis_overlays[mob_alpha_id]
+	if(!overlay)
+		overlay = _create_mob_alpha(FLY_LAYER, target_mob.plane, target_mob.dir, alpha, add_appearance_flags=RESET_ALPHA, invisibility=invisibility)
+		mob_alpha_vis_overlays[mob_alpha_id] = overlay
+		overlay.mob_alpha_id = mob_alpha_id
+		overlay.mob_owner_ref = "[REF(target_mob)]"
+	overlay.cut_overlays()
+	overlay.add_overlay(target_mob)
+
+	if(overlay in thing.vis_contents) // don't do this again
+		return overlay.mob_owner_ref
+
+	overlay.use_count++
+	thing.vis_contents += overlay
+
+	// copy-pasta from the above
+	if(!isatom(thing))
+		return overlay.mob_owner_ref
+
+	if(!thing.managed_vis_overlays)
+		thing.managed_vis_overlays = list(overlay)
+		// RegisterSignal(thing, COMSIG_ATOM_DIR_CHANGE, PROC_REF(rotate_vis_overlay))
+		// mob alpha overlay doesn't support dir rotation
+	else
+		thing.managed_vis_overlays += overlay
+
+	return overlay.mob_owner_ref
+
+/datum/controller/subsystem/vis_overlays/proc/remove_mob_alpha(atom/movable/thing, list/exception_mobs=list())
+	for(var/obj/effect/overlay/vis/mob_alpha/overlay in thing.managed_vis_overlays)
+		if(overlay.mob_owner_ref in exception_mobs)
+			continue
+		thing.vis_contents -= overlay
+		overlay.use_count--
+		if(overlay.use_count) // only remove it when it's used in nowhere
+			continue
+		mob_alpha_vis_overlays -= overlay.mob_alpha_id
+		if(isatom(thing))
+			thing.managed_vis_overlays -= overlay
+		qdel(overlay)
+
+	if(isatom(thing) && !length(thing.managed_vis_overlays))
+		thing.managed_vis_overlays = null
+		// UnregisterSignal(thing, COMSIG_ATOM_DIR_CHANGE)
```

Comment:
Why keep random commented code around?

Reply:
telling coders that it's the same copy-pasta code above, but can remove it anyway.